gitlabform/processors/project/protected_environments.py

```diff
@@ -0,0 +1,34 @@
+from gitlabform.gitlab import GitLab
+from gitlabform.processors.defining_keys import Key, And
+from gitlabform.processors.multiple_entities_processor import MultipleEntitiesProcessor
+from gitlabform.processors.util.difference_logger import DifferenceLogger
+
+
+class ProtectedEnvironmentsProcessor(MultipleEntitiesProcessor):
+    """https://docs.gitlab.com/ee/api/protected_environments.html#protect-repository-environments"""
+
+    def __init__(self, gitlab: GitLab):
+        super().__init__(
+            "protected_environments",
+            gitlab,
+            list_method_name=gitlab.list_protected_environments,
+            add_method_name=gitlab.protect_a_repository_environment,
+            edit_method_name=gitlab.update_a_protected_environment,
+            delete_method_name=gitlab.unprotect_environment,
+            # TODO: I had to define "name" inside the cfg, it should get the section's name as the entity name, e.g.
+            # protected_environments:
+            #   foo:
+            #     name: "foo" <- This is redundant
+            #
+            defining=Key("name"),
+            required_to_create_or_update=And(Key("name"), Key("deploy_access_levels")),
+        )
+
+    def _print_diff(self, project_or_project_and_group: str, entity_config: dict):
+        # TODO: yeah... I didn't get how this is supposed to work :-(
+        # Should I receive (from the super class) the "live" cfg from Gitlab ?
```

Comment:
Don't remember also. 😅 I will get back to you on this in a day or two.

Reply:
No, this is not magic. ;) You would have to implement getting the entities from GitLab. See how it's done for a SingleEntityProcessor here https://github.com/gitlabform/gitlabform/blob/16af88b5d67faaf67a7cb90640b977168820c9c8/gitlabform/processors/single_entity_processor.py#L50-L58.

But I wouldn't worry about this for now, it's fine to not implement it.

This feature should be added globally and implemented differently to make more sense. See also #290.